src/token_contract/src/main.nr

```diff
@@ -402,6 +252,46 @@ pub contract Token {
         _finalize_transfer_to_private(from, amount, hiding_point_slot, &mut context, storage);
     }
 
+    #[public]
+    #[view]
+    fn balance_of_public(owner: AztecAddress) -> U128 {
+        storage.public_balances.at(owner).read()
+    }
+
+    #[public]
+    #[view]
+    fn total_supply() -> U128 {
+        storage.total_supply.read()
+    }
+
+    /// Finalizes a mint of token `amount` to a private balance of `to`. The mint must be prepared by calling
+    /// `prepare_private_balance_increase` first and the resulting
+    /// `hiding_point_slot` must be passed as an argument to this function.
+    ///
+    /// Note: This function is only an optimization as it could be replaced by a combination of `mint_to_public`
+    /// and `finalize_transfer_to_private`. It is however used very commonly so it makes sense to optimize it
+    /// (e.g. used during token bridging, in AMM liquidity token etc.).
+    #[public]
+    fn finalize_mint_to_private(amount: U128, hiding_point_slot: Field) {
+        _finalize_mint_to_private(amount, hiding_point_slot, &mut context, storage);
+    }
```

Comment:
this is Mintable specific call, we should either deprecate it, or move it to a later section at the bottom of the file